### DISCUSSION TITLE
selection vertical per cells for horizontal lecture

### DISCUSSION THREAD

**Discussion by Larbo53** [jsvine/pdfplumber#470] [Q&A]
Hi,

I have selected vertically each column with the coordinates, and I want the horizontal reading for a return by row with the selected columns.
How do I set up pdfplumber to do this?
Can you give me an example?
Is this possible for an extraction of several tables in a pdf document?
Thanks for your feedback.

**Comment by jsvine** [comment]
Hello @Larbo53, and thanks for your interest in this library. Unfortunately, I don't fully understand this question. Could you share a specific PDF and more details about what you are trying to achieve? In addition to the PDF itself, annotated screenshots would help.

**Reply by Larbo53** [reply]
hi,
the pdf source : see attach file
[facture_F210505421.pdf](https://github.com/jsvine/pdfplumber/files/6824061/facture_F210505421.pdf)


here is my code

import pdfplumber
import pandas as pd
from pathlib import Path
from decimal import *

path="/Users/yves/Downloads"
fname = path+("/facture_F210505421.pdf")

vertical_lines = [Decimal('37.53'),Decimal('140.25'),Decimal('352.7'),Decimal('415.7'),Decimal('453.2'),Decimal('517')]
    
table_settings ={"horizontal_strategy": "lines","vertical_strategy": "explicit","explicit_vertical_lines": vertical_lines,"keep_blank_chars": True,"intersection_y_tolerance": 5}

pdf=pdfplumber.open(fname) 
page = pdf.pages[0]
df=page.extract_table(table_settings)




the résult

3                                  Origine: Bon de livr
4                                           Désignation
5                                    EAN3770003262466 -
6                                    EAN3770003262527 -
7                                    EAN3770003262503 -
8                                                      
9                                                      
10                               Conditions de paiement
11                          Echéance : 185,85 € à 30 jo
12    Modalités: Pénalités de reta\nforfaitaire de 4...
13                                                     
14                                              3 COLIS
15                                                     
16                                                 None
17                                                 None
18                                                 None
Name: 0, dtype: object
0                                                  None
1                                                  None
2                                                  None
3                        aison BL2105-27033 du 18/05/21
4                                                      
5                       AB D'AIG - MIEL DE TILLEUL 250G
6                     AB D'AIG - MIEL DE PRINTEMPS 250G
7                                  MIEL DE CAROTTE 250G
8                                                      
9                                                      
10                                                     
11               urs net (le 18 juin 2021) par virement
12    rd : selon articles L.441-10 et suivants du co...
13    Montant t\nCode TVA\nA\nQuantité to\nMontant t...
14                                                 None
15                                                 None
16                                                 None
17                                                 None
18                                                 None
Name: 1, dtype: object
0                             de facturation
1           CASTEL MÉN\ne de Ménilmon\nParis
2                                           
3                                           
4                             Prix unit\n€HT
5                                       4,79
6                                       4,39
7                                       5,50
8                                           
9                                           
10                                          
11                                          
12    e, taux appliqué :\nicle D. 441-5 du c
13                                          
14                            otal lignes HT
15                              %\n5.5\ntale
16                                  otal TTC
17                              yer\nNF203 :
18                                          

thanks a lot for your help

**Comment by Larbo53** [comment]
the desired result is attached 

Thanks à lot
[result-hoped.pdf](https://github.com/jsvine/pdfplumber/files/6824759/result-hoped.pdf)

**Comment by samkit-jain** [comment]
Hi @Larbo53 Assuming that your question is regarding how to correctly extract the table mentioned using explicit vertical lines, you have 2 options
1. Crop the page so that the cropped page contains just the table and then run the table extraction. For the PDF attached, you'll notice that if you extract tables using the default lines-lines settings, it will give you 5 tables. The second table is the table of interest and using the `.find_tables()` method, you can get the bounding box of the table and then crop the page to those coordinated and then run the table extraction using the explicit vertical lines strategy.
2. Extract the table the same way you have mentioned in your code but as a post processing step, drop all the cells that have `None` in them and then, keep only those rows that have the required number of columns in them.

**Comment by Larbo53** [comment]
hi Samkit-jian,
I have already found the solution. Thanks for your answer.
Thank you for your answer.
